packages/starlight/index.ts

```diff
@@ -43,11 +43,15 @@ export default function StarlightIntegration({
 					injectRoute({
 						pattern: '404',
 						entrypoint: '@astrojs/starlight/404.astro',
+						// Ensure page is pre-rendered even when project is on server output mode
+						prerender: true,
```

Comment:
Mainly out of curiosity, why do we need both this AND the `export const prerender`? Seems like an API smell in Astro to me, so if these are both required I’d love to understand why.

Reply:
It is something that I noticed while working on the routing logic. Different parts of Astro core have different behavior for prerendered and on-demand routes, but those different parts don't all have access to both those methods of defining whether something is prerendered.

For example, the conflict warning about on-demand routes with the same dynamic segments runs before the file is read, so it only knows about this prerender parameter here. Another part, that generates another warning about not matching a static path in the dev server, only knows about the prerender constant from the file, not the value passed here.

So to have a consistent behavior everywhere this needs to be passed in both places.

I haven't found anything but warning that check only one of those, but I haven't checked too deeply either.

Reply:
This _could_ also relate to endpoints not being prerendered if just this flag is passed in (which I mentioned here: https://github.com/withastro/astro/issues/7721#issuecomment-1847763893). But that is mostly speculation on my part, I haven't actually tried to debug that at all.